simphony/io/h5_cuds.py

```diff
@@ -152,6 +155,47 @@ def add_mesh(self, mesh):
         self._file.flush()
         return m
 
+    def add_lattice(self, lattice, record=None):
```

Comment:
The use-case with a problem is the following: We have a 3D Lattice, that is used to hold in-memory data of some Lattice-Boltzmann simulation and it containes for each LatticeNode CUBA-data for density, material_id, velocity, and a 19-dimensional 64bit float vector, that is used internally in the simulation engine. Now, only density, material_id and velocity are requested to be saved into a H5CUDS-file. 

Of course, this could be done using the FileLattice constructor directly with a specific custom record class, and then iterate over the Lattice and then update the FileLattice nodes. However, this would separate full and partial saving of lattices under different methods (add_lattice and FileLattice.__init__ + etc.) I hope that we can somehow allow partial adding of the CUBA-keys with add_lattice so that it would still work similarly to the other CUDS-types and their related H5CUDS-methods. I also think that partial adding, in the case of Lattice-Boltzmann, will be the more often used scenario.